### ISSUE TITLE
*.ts *.mkv video file be application/octet-stream when i using direct upload.

### ISSUE THREAD

**Issue by x1wins** [rails/rails#39944]
I checked tree point and found that content-type was application/octet-stream or video/MP2T
1. direct upload log
```
web_1      | Processing by ActiveStorage::DirectUploadsController#create as JSON
web_1      |   Parameters: {"blob"=>{"filename"=>"sample_1920x1080.ts", "content_type"=>"application/octet-stream", "byte_size"=>15074780, "checksum"=>"VeJxcPFo/qs9pPbwfSFdQw=="}, "direct_upload"=>{"blob"=>{"filename"=>"sample_1920x1080.ts", "content_type"=>"application/octet-stream", "byte_size"=>15074780, "checksum"=>"VeJxcPFo/qs9pPbwfSFdQw=="}}}
web_1      |    (0.9ms)  BEGIN
web_1      |   ActiveStorage::Blob Create (0.6ms)  INSERT INTO "active_storage_blobs" ("key", "filename", "content_type", "byte_size", "checksum", "created_at") VALUES ($1, $2, $3, $4, $5, $6) RETURNING "id"  [["key", "kzatm1dk3ke5dkxrwbgajbcpxnfb"], ["filename", "sample_1920x1080.ts"], ["content_type", "application/octet-stream"], ["byte_size", 15074780], ["checksum", "VeJxcPFo/qs9pPbwfSFdQw=="], ["created_at", "2020-07-28 06:40:19.234414"]]
```
2. original file content-type
```
$ file --mime-type /Users/xxx/Downloads/sample_1920x1080.ts 
/Users/xxx/Downloads/sample_1920x1080.ts: application/octet-stream
```
3. irb
```
$ gem install mime-types
irb(main):001:0> require 'mime-types'
=> true
irb(main):002:0> MIME::Types.type_for("/Users/xxx/Downloads/sample_1920x1080.ts").first.content_type
=> "video/MP2T"
```

### Steps to reproduce
1. Download sample video *.ts file at https://filesamples.com/formats/ts
2. direct upload implement or install ```git clone https://github.com/x1wins/CW-OVP.git```
https://github.com/x1wins/CW-OVP/issues/5

### Expected behavior
*.ts content-type must be ```video/mp2t```
*.mkv content-type must be ```video/x-matroska```
*.ts *.mkv are video file
they have to be content_type=>"video/**"

### Actual behavior
*.ts *.mkv extension file uploaded with content_type=>"application/octet-stream" when i using direct upload.
i know default content_type value is "application/octet-stream" in active storage

### System configuration
**Rails version**: 6.0.3

**Ruby version**: 2.6.3

**Comment by jonathanhefner** [comment]
I was not able to reproduce `"content_type"=>"application/octet-stream"`.  However, I did see that the `.ts` extension is associated with the `text/vnd.trolltech.linguist` (aka `text/vnd.qt.linguist`) MIME type by default.  When not using direct uploads, Rails is able to inspect the file and correct this MIME type to `video/mp2t`, but when using direct uploads Rails must rely on metadata only.

Does the error in x1wins/CW-OVP#5 still occur with an `.mts` file extension?

**Comment by x1wins** [comment]
> I was not able to reproduce `"content_type"=>"application/octet-stream"`. However, I did see that the `.ts` extension is associated with the `text/vnd.trolltech.linguist` (aka `text/vnd.qt.linguist`) MIME type by default. When not using direct uploads, Rails is able to inspect the file and correct this MIME type to `video/mp2t`, but when using direct uploads Rails must rely on metadata only.
> 
> Does the error in [x1wins/CW-OVP#5](https://github.com/x1wins/CW-OVP/issues/5) still occur with an `.mts` file extension?

yes there is still occur error with ```.mts .ts .mkv``` when using direct upload.
you can reproduce if you get CW-OVP 
### Steps to reproduce
1. ```git clone https://github.com/x1wins/CW-OVP.git```
2. ```cd ./CW-OVP```
3. ```git reset --hard 8e5e52f``` now i deleted content_type validate code at lastest develop branch
4. ```docker-compose up --build -d```
5. ```docker-compose run --no-deps web bundle exec rails webpacker:install```
6. ```docker-compose run --no-deps web bundle exec rake db:migrate```
7. and open web browser with ```http://localhost:3000/encodes/new```
![image](https://user-images.githubusercontent.com/982291/88851820-35752c00-d1a2-11ea-9fe0-00ac463499a7.png)

- sample video download
   * mts download https://filesamples.com/formats/mts
   * ts download https://filesamples.com/formats/ts
   * mkv download https://filesamples.com/formats/mkv

i did try direct upload with .mts
i got below log file. that is same ```*.ts *.mkv``` file's content_type value
there is ```"filename"=>"sample_1280x720.mts", "content_type"=>"application/octet-stream"```

```
web_1      | Started POST "/rails/active_storage/direct_uploads" for 172.22.0.1 at 2020-07-29 20:46:06 +0000
web_1      | Cannot render console from 172.22.0.1! Allowed networks: 127.0.0.0/127.255.255.255, ::1
web_1      | Processing by ActiveStorage::DirectUploadsController#create as JSON
web_1      |   Parameters: {"blob"=>{"filename"=>"sample_1280x720.mts", "content_type"=>"application/octet-stream", "byte_size"=>6734724, "checksum"=>"9mY7r+InO/1F68TsNEksRQ=="}, "direct_upload"=>{"blob"=>{"filename"=>"sample_1280x720.mts", "content_type"=>"application/octet-stream", "byte_size"=>6734724, "checksum"=>"9mY7r+InO/1F68TsNEksRQ=="}}}
```

**Comment by x1wins** [comment]
> Rails is able to inspect the file and correct this MIME type to `video/mp2t`, but when using direct uploads Rails must rely on metadata only.

How can i change direct upload for that like a Rails inspect the file and how make correct MIME type `video/mp2t`?
if not use direct upload. ```*.ts,*.mkv, *.mts``` content type value stored ```content-type : video/**``` in active storage table. this is correct and I won expected value. 
but ```"content_type"=>"application/octet-stream"``` only occurd when using direct upload

**Comment by jonathanhefner** [comment]
> How can i change direct upload for that like a Rails inspect the file and how make correct MIME type `video/mp2t`?

It is up to the browser to submit the correct MIME type.  (Note that the browser could guess wrong based on the file extension, or that a malicious user could intentionally submit an incorrect MIME type.)  You can change the recorded MIME type using `ActiveStorage::Blob#content_type=`, but that will not fix your validation.  

What browser are you using that is submitting `"content_type"=>"application/octet-stream"`?

If incorrect MIME types are something you have to account for, you might trying using `MimeMagic` directly:

```ruby
MimeMagic.by_path(filename).video?
```

And you can configure which extensions `MimeMagic` recognizes (see [documentation](https://www.rubydoc.info/github/minad/mimemagic/MimeMagic)).

**Comment by georgeclaghorn** [comment]
> When not using direct uploads, Rails is able to inspect the file and correct this MIME type to video/mp2t, but when using direct uploads Rails must rely on metadata only.

We identify directly-uploaded blobs [when they’re attached](https://github.com/rails/rails/blob/b39ea7cc3662345ad8efbe496db5ae594449f85f/activestorage/app/models/active_storage/attachment.rb#L41-L43). @x1wins, does the blob’s `content_type` get corrected when you attach it to something?

**Comment by x1wins** [comment]
> > How can i change direct upload for that like a Rails inspect the file and how make correct MIME type `video/mp2t`?
> 
> It is up to the browser to submit the correct MIME type. (Note that the browser could guess wrong based on the file extension, or that a malicious user could intentionally submit an incorrect MIME type.) You can change the recorded MIME type using `ActiveStorage::Blob#content_type=`, but that will not fix your validation.
> 
> What browser are you using that is submitting `"content_type"=>"application/octet-stream"`?
> 
> If incorrect MIME types are something you have to account for, you might trying using `MimeMagic` directly:
> 
> ```ruby
> MimeMagic.by_path(filename).video?
> ```
> 
> And you can configure which extensions `MimeMagic` recognizes (see [documentation](https://www.rubydoc.info/github/minad/mimemagic/MimeMagic)).

i used chrome web browser
and then i tried ```MimeMagic```
but i got false from MimeMagic
```
irb(main):001:0> require 'MimeMagic'
=> true
irb(main):002:0> filename="/Users/xxx/Downloads/sample_1920x1080_1.ts"
=> "/Users/rhee/Downloads/sample_1920x1080_1.ts"
irb(main):003:0> MimeMagic.by_path(filename).video?
=> false
```

**Comment by x1wins** [comment]
> > When not using direct uploads, Rails is able to inspect the file and correct this MIME type to video/mp2t, but when using direct uploads Rails must rely on metadata only.
> 
> We identify directly-uploaded blobs [when they’re attached](https://github.com/rails/rails/blob/b39ea7cc3662345ad8efbe496db5ae594449f85f/activestorage/app/models/active_storage/attachment.rb#L41-L43). @x1wins, does the blob’s `content_type` get corrected when you attach it to something?

it's working well and get content-type ```video/mp4``` when i use direct upload with ``` *.mp4 *.mov ``` video file
below logs is ```*.mp4``` direct upload logs. there are ```"content_type"=>"video/mp4"```.  this is correct

```
web_1      | Started POST "/rails/active_storage/direct_uploads" for 172.21.0.1 at 2020-08-06 03:54:36 +0000
web_1      | Cannot render console from 172.21.0.1! Allowed networks: 127.0.0.0/127.255.255.255, ::1
web_1      | Processing by ActiveStorage::DirectUploadsController#create as JSON
web_1      |   Parameters: {"blob"=>{"filename"=>"sample-mp4-file.mp4", "content_type"=>"video/mp4", "byte_size"=>10546620, "checksum"=>"UCGzt8QCRo1bAYqLSitEig=="}, "direct_upload"=>{"blob"=>{"filename"=>"sample-mp4-file.mp4", "content_type"=>"video/mp4", "byte_size"=>10546620, "checksum"=>"UCGzt8QCRo1bAYqLSitEig=="}}}
web_1      |    (0.5ms)  BEGIN
web_1      |   ActiveStorage::Blob Create (12.0ms)  INSERT INTO "active_storage_blobs" ("key", "filename", "content_type", "byte_size", "checksum", "created_at") VALUES ($1, $2, $3, $4, $5, $6) RETURNING "id"  [["key", "umhykjzpbn1bly3nkm2bl5erqkx2"], ["filename", "sample-mp4-file.mp4"], ["content_type", "video/mp4"], ["byte_size", 10546620], ["checksum", "UCGzt8QCRo1bAYqLSitEig=="], ["created_at", "2020-08-06 03:54:36.182491"]]
```

**Comment by jonathanhefner** [comment]
@x1wins This issue should be fixed in Rails 6.1 by #40013.  Thank you for calling attention to it!
